internal/databases/postgres/wal_push_handler.go

```diff
@@ -82,7 +87,24 @@ func uploadWALFile(uploader *WalUploader, walFilePath string, preventWalOverwrit
 		return errors.Wrapf(err, "upload: could not open '%s'\n", walFilePath)
 	}
 	err = uploader.UploadWalFile(walFile)
-	return errors.Wrapf(err, "upload: could not Upload '%s'\n", walFilePath)
+
+	if err != nil {
+		return errors.Wrapf(err, "upload: could not Upload '%s'\n", walFilePath)
+	}
+
+	// rename WAL status file ".ready" to ".done" if requested
+	if ReadyRename && err == nil {
```

Comment:
Maybe it's better to do this in ArchiveStatusManager? Or somewhere near...

Reply:
Good idea, it should be cleaner that way

Reply:
done